engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -80,24 +80,36 @@ def prepare_scc_announce_request(uri_path, auth, params)
       # to SCC.
       # SCC will make sure to handle the data correctly. This removes the need
       # to adapt here if information send by the client changes.
-      scc_request.body = {
+      scc_req_body = {
         hostname: params['hostname'],
         hwinfo: params['hwinfo'],
-        byos: true
-      }.to_json
+        byos_mode: params['proxy_byos_mode']
+      }
+      # when system is payg, we do not know whether it's hybrid or not
+      # we send the login and password information to skip the validation
+      # on the SCC side, that info is enough to validate the product later on
+      # if the system is, in fact, hybrid
```

Comment:
This comment reads like as if PAYG systems will announce to SCC. That should not happen. Only in the case when the system is detected as byos or hybrid it should get announced to SCC.

Reply:
we don't know if PAYG is hybrid until after validate the registration code

Reply:
I think it's just the wording here. I saw below that only activations attempts with regcode would get announced to SCC, that's fine.